JavaScript/JavaScriptSDK.Tests/CheckinTests/TelemetryContext.tests.ts

```diff
@@ -68,6 +74,155 @@ class TelemetryContextTests extends TestClass {
                 Assert.equal(true, env.tags[contextKeys.sessionIsFirst], "session.isFirst");
             }
         });
+
+        this.testCase({
+            name: "TelemetryContext: properties initialized correctly",
+            test: () => {
+                // act
+                var telemetryContext = new Microsoft.ApplicationInsights.TelemetryContext(this._config);
+
+                // verify
+                Assert.deepEqual(this._config.properties(), telemetryContext.properties);
+            }
+        });
+
+        this.testCase(
+            {
+                name: "TelemetryContext: custom properties applied if telemetry item's properties are undefined",
+                test: () => {
+                    // setup
+                    this._telemetryContext.properties = { prop1: "val1" };
+                    var eventEnvelope = this.getTestEventEnvelope(undefined);
+                    // act
+                    (<any>this._telemetryContext)._applyCustomProperties(eventEnvelope);
```

Comment:
Can we test at the public interface level and call `track` instead of the private implementation level (`_applyCustomProperties`)?